domain/secretbackend/service/service.go

```diff
@@ -0,0 +1,580 @@
+	reveal, all bool, names ...string,
+) ([]*SecretBackendInfo, error) {
+	backends, err := s.st.ListSecretBackends(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	backendInfos := make([]*SecretBackendInfo, 0, len(backends))
+	for _, b := range backends {
+		backendInfos = append(backendInfos, &SecretBackendInfo{
+			SecretBackend: coresecrets.SecretBackend{
+				ID:                  b.ID,
+				Name:                b.Name,
+				BackendType:         b.BackendType,
+				TokenRotateInterval: b.TokenRotateInterval,
+				Config:              convertConfigToAny(b.Config),
+			},
+		})
+	}
+	// If we want all backends, include those which are not in use.
+	if all {
+		// The internal (controller) backend.
+		backendInfos = append(backendInfos, &SecretBackendInfo{
+			SecretBackend: coresecrets.SecretBackend{
+				ID:          s.controllerUUID,
+				Name:        juju.BackendName,
+				BackendType: juju.BackendType,
+			},
+		})
+		m, err := s.st.GetModel(ctx, modelUUID)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if m.Type == coremodel.CAAS {
+			// The kubernetes backend.
+			k8sConfig, err := getK8sBackendConfig(cloud, cred)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			k8sBackend := &SecretBackendInfo{
+				SecretBackend: coresecrets.SecretBackend{
+					ID:          m.ID.String(),
+					Name:        kubernetes.BuiltInName(m.Name),
+					BackendType: kubernetes.BackendType,
+					Config:      k8sConfig.Config,
+				},
+				// TODO: implement secret count for secret backend.
+				// For now, we just set it to 1 to indicate that the backend is in use.
+				NumSecrets: 1,
+			}
+			// For local k8s secrets, corresponding to every hosted model,
+			// do not include the result if there are no secrets.
+			if k8sBackend.NumSecrets > 0 || !all {
+				backendInfos = append(backendInfos, k8sBackend)
+			}
+		}
+	}
+	wanted := set.NewStrings(names...)
+	for i := 0; i < len(backendInfos); {
+		b := backendInfos[i]
+		if !wanted.IsEmpty() && !wanted.Contains(b.Name) {
+			backendInfos = append(backendInfos[:i], backendInfos[i+1:]...)
+			continue
+		} else {
+			i++
+		}
+		p, err := s.registry(b.BackendType)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		b.Status = status.Active.String()
+		if b.BackendType != juju.BackendType && b.BackendType != kubernetes.BackendType {
+			if err := pingBackend(p, b.Config); err != nil {
+				b.Status = status.Error.String()
+				b.Message = err.Error()
+			}
+		}
+		if len(b.Config) == 0 {
+			continue
+		}
+		configValidator, ok := p.(provider.ProviderConfig)
+		if !ok {
+			continue
+		}
+		for n, f := range configValidator.ConfigSchema() {
+			if f.Secret && !reveal {
+				delete(b.Config, n)
+			}
+		}
+	}
+	return backendInfos, nil
+}
+
+// CheckSecretBackend checks the secret backend for the given backend ID.
+func (s *Service) CheckSecretBackend(ctx context.Context, backendID string) error {
```

Comment:
Can make the method name include what is being checked, eg

CheckSecretBackendAlive()

or something

or even PingSecretBackend()